hack/verify-contextual-logging.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+LOG_FILE="./${SCRIPT_ROOT}/hack/logcheck.out"
+work_file="$(mktemp)"
+path_prefix="$(dirname $(cd ${SCRIPT_ROOT}; pwd))"
+
+function cleanup() {
+    rm -rf "${work_file}"
+}
+
+trap cleanup EXIT
+
+# pkg/apis is a separate module, so check that in addition to our root packages
+if logcheck -check-contextual "${SCRIPT_ROOT}"/pkg/... >"${work_file}" 2>&1 && \
+   $(cd "${SCRIPT_ROOT}"/pkg/apis; logcheck -check-contextual ./... >>"${work_file}" 3>&1); then
+
+    echo "[ERROR] logcheck found no errors, but expected at least $(wc -l < ${LOG_FILE})"
+    exit 1
+fi
+
+# Normalize paths so we don't generate diffs based only on user directory mismatches
+sed -i'' -e "s,${path_prefix},,g" "${work_file}"
```

Comment:
`sed -i''` was somehow broken for me - could you look at the hack I put into `hack/bump-k8s.sh` to work around it?

Reply:
Out of curiosity, are you on Linux? My understanding was `-i''` was better supported between macOS/GNU sed, but apparently that's not true.

Reply:
Yes, I'm on Linux